### ISSUE TITLE
TF docker error tensorflow/serving:1.13.0-devel-gpu ImportError: libcublas.so.9.0

### ISSUE THREAD

**Issue by alexcpn** [tensorflow/serving#1316]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow Serving installed from (source or binary)**:  Binary Docker
- **TensorFlow Serving version**:
 tensorflow/serving:1.13.0-devel-gpu 

### Describe the problem
The tensorflow in the continer does not load.Gives  error - I`mportError: libcublas.so.9.0: cannot open shared object file: No such file or directory`

### Exact Steps to Reproduce
```
docker run -it tensorflow/tf-serving:1.13.0-devel-gpu
invoke python shell
import tensorflow

```
### Source code / logs
```
>>> import tensorflow
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
ImportError: libcublas.so.9.0: cannot open shared object file: No such file or directory

```
**Solution**
Reinstalling tensorflow-gpu version 1.13 solves the problem
pip install tensorflow-gpu==1.13.1

  Found existing installation: tensorflow-gpu 1.12.0
    Uninstalling tensorflow-gpu-1.12.0:
      Successfully uninstalled tensorflow-gpu-1.12.0
Successfully installed tensorboard-1.13.1 tensorflow-estimator-1.13.0 tensorflow-gpu-1.13.1

**Comment by hgadig** [comment]
Could you please provide cuda and cudnn versions?

**Comment by netfs** [comment]
thanks for reporting the issue!

this is a bug in our release build process. specifically when we build tensorflow-serving-api PIP package we pick the latest TF pip package:https://github.com/tensorflow/serving/blob/28da549e382fa027897886afec83fca6e92da33c/tensorflow_serving/tools/pip_package/setup.py#L49 which may or may-not be inline with TF Serving version. depends on when these builds start. its possible TF PIP build is not complete yet for say 1.13.0 version, when TF Serving docker build attempts to install tensorflow-serving-api-gpu package, and picks up latest TF package (1.12.0 in this case))

**Comment by netfs** [comment]
OK so this is happening on tensorflow/serving:nightly-devel-gpu also where TF 1.13.1 is installed within the docker image. installing TF nightly resolves the issue.

**Comment by aaroey** [comment]
@christisg may have some thoughts on this.

**Comment by aaroey** [comment]
@chsigg may have an idea about this.

**Comment by netfs** [comment]
folk folks attempting to reproduce this (needs docker installed) with latest bits:
```
$ docker pull tensorflow/serving:nightly-devel-gpu

$ docker run -it tensorflow/serving:nightly-devel-gpu

root@1567a50c6017:/tensorflow-serving#  pip list | grep tensorflow
tensorflow-estimator       1.13.0   
tensorflow-gpu             1.13.1   
tensorflow-serving-api-gpu 0.0.0

root@1567a50c6017:/tensorflow-serving# python -c 'import tensorflow'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
ImportError: libcuda.so.1: cannot open shared object file: No such file or directory


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/errors

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

root@1567a50c6017:/tensorflow-serving# find / -name "libcuda.so*"
/usr/local/cuda-10.0/compat/libcuda.so.1
/usr/local/cuda-10.0/compat/libcuda.so
/usr/local/cuda-10.0/compat/libcuda.so.410.104
/usr/local/cuda-10.0/targets/x86_64-linux/lib/stubs/libcuda.so
```

**Comment by netfs** [comment]
@alexcpn can you try by adding --runtime=nvidia to your docker command line when using gpu docker images. this works (without any failures for me) for the `nightly-devel-gpu` version:

$ docker run --runtime=nvidia --rm -it tensorflow/serving:nightly-devel-gpu  python -c 'import tensorflow'

thanks to @gunan for noticing the missing `--runtime=nvidia` above.

**Comment by alexcpn** [comment]
I dont have access to the remote machine now, but I usually give that  (--runtime=nviidia) Can you try with the exact release tf-serving:1.13.0-devel-gpu, instead of the latest. Also you can close it if it is fixed in latest. Thanks

**Comment by netfs** [comment]
yes this is still an issue with our 1.13 devel-gpu docker images (as it has TF 1.12 installed).

this does not happen with nightly-devel-gpu (noted above).

i will keep this issue open and ensure upcoming 1.14 release does not have this issue.

thanks for reporting it.

**Comment by netfs** [comment]
verified that this is fixed in the latest 1.14.0-rc0 docker image:

```
$ docker run -it tensorflow/serving:1.14.0-rc0-devel-gpu pip list | grep tensorflow
tensorflow-gpu             1.14.0rc0           
tensorflow-serving-api-gpu 1.14.0rc0
$ docker run --runtime=nvidia --rm -it tensorflow/serving:1.14.0-rc0-devel-gpu python -c 'import tensorflow as tf; print(tf.__version__)'
1.14.0-rc0
$
```
